tudatBundle/pagmo2/include/pagmo/algorithms/giACOmo.hpp

```diff
@@ -0,0 +1,600 @@
+            pagmo_throw(std::invalid_argument,
+                        "The Pareto precision parameter must be in [0, 1[, while a value of "
+                            + std::to_string(epsilon) + " was detected");
+        }
+
+
+    }
+
+    /// Algorithm evolve method (juice implementation of the algorithm)
+    /**
+     *
+     * Evolves the population for the requested number of generations.
+     *
+     * @param pop population to be evolved
+     * @return evolved population
+     * @throw std::invalid_argument if pop.get_problem() is stochastic, single objective or has non linear constraints.
+     * If \p int_dim is larger than the problem dimension. If the population size is smaller than 5 or not a multiple of
+     * 4.
+     */
+    population evolve(population pop) const
+    {
+        // We store some useful variables
+        const auto &prob = pop.get_problem(); // This is a const reference, so using set_seed for example will not be
+                                              // allowed
+        auto dim = prob.get_nx();             // This getter does not return a const reference but a copy
+        auto NP = pop.size();
+
+        const auto bounds = prob.get_bounds();
+        const auto &lb = bounds.first;
+        const auto &ub = bounds.second;
+
+        //Note that the number of equality and inequality constraints has to be set up manually in the problem definition,
+        //otherwise PaGMO assumes that there aren't any
+        auto NIC = prob.get_nic(); //number of inequality constraints
+        auto NEC = prob.get_nec(); //number of equality constraints
+
+        auto fevals0 = prob.get_fevals(); // discount for the fevals already made
+        unsigned count = 1u;          // regulates the screen output
+
+        // PREAMBLE-------------------------------------------------------------------------------------------------
+        // We start by checking that the problem is suitable for this
+        // particular algorithm.
+
+        if (!NP) {
+            pagmo_throw(std::invalid_argument, get_name() + " cannot work on an empty population");
+        }
+        if (prob.is_stochastic()) {
+            pagmo_throw(std::invalid_argument,
+                        "The problem appears to be stochastic " + get_name() + " cannot deal with it");
+        }
+        if (prob.get_nc() != 0u) {
+            pagmo_throw(std::invalid_argument, "Non linear constraints detected in " + prob.get_name() + " instance. "
+                                                   + get_name() + " cannot deal with them.");
+        }
+        if (m_gen == 0u){
+            return pop;
+        }
+
+
+
+        // ---------------------------------------------------------------------------------------------------------
+
+        // No throws, all valid: we clear the logs
+        m_log.clear();
+
+        // Declarations
+
+
+        // Main ACO loop over generations:
+        for (decltype(m_gen) gen = 1u; gen <= m_gen; gen++) {
```

Comment:
The log is the last thing to do. I would remove this part for the time being, when evolve is finished we can talk about what to log.

Reply:
Sure, I will remove it.